### PR TITLE
update lmod and modules for fram

### LINKED ISSUE #56: Update Fram issue after updates
Fram has been updated which requires an update to its entry in the config_machines.xml file.

### PR DESCRIPTION
**Pull request by mvdebolskiy**
Changed modules to newer the ones built with newer toolchains. Changed the path to lmod languages. Needed for model to work on the updated Fram.

Test suite: Tested by running a 5day test using N1850 compset and f19_tn14 grid. 

Fixes #56 

User interface changes?: N

Update gh-pages html (Y/N)?: N

### CODE REVIEW

**Comment by gold2718** [comment]
@mvdebolskiy, Can you add a brief note as to how you tested this (e.g., ran an xxx case)?

**Review by gold2718 — APPROVED** [review]
:+1:

**Comment by mvdebolskiy** [comment]
Tested with 5day run using `N1850` compset and  `f19_tn14` grid. Any other comset I should try?

File: config/cesm/machines/config_machines.xml [hunk 1]
```diff
@@ -2260,6 +2260,7 @@ This allows using a different mpirun command to launch unit tests
 
   <machine MACH="fram">
     <DESC>Lenovo NeXtScale M5, 32-way nodes, dual 16-core Xeon E5-2683@2.10GHz, 64 GiB per node, os is Linux, batch system is SLURM</DESC>
+    <NODENAME_REGEX>.*.?fram\d?.sigma2.no</NODENAME_REGEX>
```
Comment (gold2718) [inline config/cesm/machines/config_machines.xml:2263]: Thanks, forgot to ask for this.

**Comment by gold2718** [comment]
> Tested with 5day run using `N1850` compset and `f19_tn14` grid. Any other comset I should try?

No, I think this type of change either works or not.

**Review by mvertens — CHANGES_REQUESTED** [review]
@mvdebolskiy - thanks so much for this.
Could we also add the path to CCSM_CPRNC and BASELINE_ROOT for both betzy and fram:
CCSM_CPRNC = /cluster/shared/noresm/tools/cprnc/cprnc
BASELINE_ROOT=/cluster/shared/noresm/noresm_baselines

**Comment by mvdebolskiy** [comment]
Done.

**Comment by mvertens** [comment]
@mvdebolskiy - thanks so much! this is super helpful.

**Review by mvertens — APPROVED** [review]

**Comment by mvdebolskiy** [comment]
Tested with `--res f19_tn14--compset N1850frc2 --pecount 128 --queue devel` on betzy. Works. Also RESUBMIT does not fail. Only weird thing is st._archive jobs take around 7 minutes. (also, had to do `NTASKS_OCN=123` since 128 does not work). Also, now you can check out small number of nodes on `preproc` queue for a really small job.

**Comment by mvertens** [comment]
@mvdebolskiy - thanks for this and for the testing. This will be super helpful in the upcoming release.

**Comment by mvdebolskiy** [comment]
Should also fix #49 for betzy. Is #49 needs to fix it on fram also?

**Comment by gold2718** [comment]
> Should also fix #49 for betzy. Is #49 needs to fix it on fram also?

Is the queue structure the same? The request I received was for Betzy since support for the devel queue had been removed for the NorESM2 release due to some problem which I understand has been resolved. Does Fram have any useful queues that are not listed in its config entry?

**Comment by mvdebolskiy** [comment]
@gold2718 the only reson we have to use `devel` and `preproc` queues on betzy is that normal queue demands the checkout of at least 4 nodes. Fran does not have this issue. I can add more queues to fram but I doubt anyone would use them to run release code.

**Pull request merged**